bsc/header.go

```diff
@@ -185,31 +186,54 @@ func Keccak256(data ...[]byte) []byte {
 }
 
 // SealHash returns the hash of a block prior to it being sealed.
-func SealHash(header *Header) (hash Hash) {
+func SealHash(header *Header, chainID *big.Int) (hash Hash) {
 	hasher := sha3.NewLegacyKeccak256()
-	encodeSigHeader(hasher, header)
+	encodeSigHeader(hasher, header, chainID)
 	hasher.Sum(hash[:0])
 	return hash
 }
 
-func encodeSigHeader(w io.Writer, header *Header) {
-	err := rlp.Encode(w, []interface{}{
-		header.ParentHash,
-		header.UncleHash,
-		header.Coinbase,
-		header.Root,
-		header.TxHash,
-		header.ReceiptHash,
-		header.Bloom,
-		big.NewInt(header.Difficulty),
-		big.NewInt(header.Number),
-		header.GasLimit,
-		header.GasUsed,
-		header.Time,
-		header.Extra[:len(header.Extra)-65], // this will panic if extra is too short, should check before calling encodeSigHeader
-		header.MixDigest,
-		header.Nonce,
-	})
+func encodeSigHeader(w io.Writer, header *Header, chainId *big.Int) {
+	var err error
+	if sdk.IsUpgrade(sdk.BEP174) {
```

Comment:
`encodeSigHeader` function is better to be a stateless function, it may not have the block height context.  Since `chainId` is a pointer, I will suggest utilize this , we encode it only when `chainId` is not nil,